### PR TITLE
Overloaded SolverManager.create() methods + move ScoreManagerTest

### PR DESCRIPTION
**Pull request by ge0ffrey**
Motivation for the overloaded SolverManager.create() methods: they make it easier to get started with OptaPlanner. And easier to explain to users how to get started with it.
Most users don't need to set a SolverManagerConfig.

### CODE REVIEW

File: optaplanner-core/src/main/java/org/optaplanner/core/api/solver/SolverManager.java [hunk 1]
```diff
@@ -51,6 +51,22 @@
...
+    /**
+     * Use a {@link SolverConfig} to build a {@link SolverManager}.
+     * <p>
+     * When using {@link ScoreManager} too, use {@link #create(SolverFactory)} instead
+     * so they reuse the same {@link SolverFactory} instance.
+     *
+     * @param solverConfig never null
+     * @return never null
+     * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+     * @param <ProblemId_> the ID type of a submitted problem, such as {@link Long} or {@link UUID}.
```
Comment (triceo) [inline optaplanner-core/src/main/java/org/optaplanner/core/api/solver/SolverManager.java:63]:
```suggestion
     * @param <ProblemId_> the ID type of a submitted problem, such as {@link Long} or {@link UUID}
```

As much as I don't like it, that's the convention.

File: optaplanner-core/src/main/java/org/optaplanner/core/api/solver/SolverManager.java [hunk 2]
```diff
@@ -68,6 +84,19 @@
         return create(SolverFactory.create(solverConfig), solverManagerConfig);
     }
 
+    /**
+     * Use a {@link SolverFactory} to build a {@link SolverManager}.
+     *
+     * @param solverFactory never null
+     * @return never null
+     * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+     * @param <ProblemId_> the ID type of a submitted problem, such as {@link Long} or {@link UUID}.
```
Comment (triceo) [inline optaplanner-core/src/main/java/org/optaplanner/core/api/solver/SolverManager.java:93]:
```suggestion
     * @param <ProblemId_> the ID type of a submitted problem, such as {@link Long} or {@link UUID}
```

**Review by triceo — APPROVED** [review]
Minor suggestions.

**Comment by ge0ffrey** [comment]
Squash or don't squash, it's good for me.

**Comment by triceo** [comment]
Jenkins retest this.

**Comment by rsynek** [comment]
There seem to be merge conflicts, would you take a look, @ge0ffrey?

**Pull request merged**